### DISCUSSION TITLE
Running Renovate against a single github repository using github actions

### DISCUSSION THREAD

**Discussion by jessehouwing** [renovatebot/renovate#20189] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest

### If you're self-hosting Renovate, select which platform you are using.

github.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

latest

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

a simple script I can add to the github actions to run renovate on a private runner, just for this repo. Which happens to be a fork.

### What you tried so far.

I created a actions workflow and added 

```
npx --autodiscover
RENOVATE_REPOSITORIES: ${{ github.repository }}
```

But that seems to still fetch my 3800 accessible repositories. Which surprised me. I expected the env var to forego fetching all my repos. It surprised me too, by the way, that it defaults to EVERYTHING and not just to the repo in my working directory. Was able to cancel the workflow in time.

Would be nice to have a `--what-if` option, or maybe it exists and I didn't find it yet.

Replaced it with

```
npx $env:REPOSITORY
REPOSITORY: ${{ github.repository }}
```

which is **MUCH** faster, since it only runs against this explicit repo. yay! But it fails because the repo is a fork. Which again surptised me. Im passing the repo **explicitly**. I'd expect renovate to honer my request. 

[I found that Renovate wants my `renovate.json` in the root of the repo and not in the `.github` folder](https://github.com/renovatebot/app-support/issues/114). And unfortunately I can't tell `npx renovate` that this is where all my configs be stored, right?

Something like

```
npx $env:REPOSITORY --config-path="./github/renovate.json"
REPOSITORY: ${{ github.repository }}
```

Or that renovate would not try to be clever and just accept that me passing in a list of hardcoded repos means I want it to scan them regardless.

Now I'm trying this:

```
npx $env:REPOSITORY --include-forks
REPOSITORY: ${{ github.repository }}
```

Which failed, cause `--include-forks` isn't a switch, but requires an explicit value, so this finally seems to work.

```
npx $env:REPOSITORY --include-forks=true
REPOSITORY: ${{ github.repository }}
```

I love the tool and what it does, but it has so many surprising behaviors and the docs, while very thorough, aren't helping me do the right thing the first time I try them. I really feel I'm stumbling into new surprises around every corner.

The Error/Info messages also don't point me to a resolution on how to fix things. I would have loved a mention to the `--include-forks` docs  in the "Repo is fork, skipping" message. The requirement of teh config file being in the root is also not present in the docs:

https://docs.renovatebot.com/configuration-options/#includeforks

### Relevant debug logs

<details><summary>Logs</summary>

```
npm WARN exec The following package was not found and will be installed: renovate@34.[12](https://github.com/jessehouwing/azure-devops-extension-tasks/actions/runs/4082851180/jobs/7037680766#step:4:13)1.0
npm WARN deprecated @npmcli/move-file@2.0.1: This functionality has been moved to @npmcli/fs
 INFO: Repository started (repository=jessehouwing/azure-devops-extension-tasks)
       "renovateVersion": "34.121.0"
 INFO: Repository is a fork and not manually configured - skipping (repository=jessehouwing/azure-devops-extension-tasks)
 INFO: Repository finished (repository=jessehouwing/azure-devops-extension-tasks)
       "cloned": false,
       "durationMs": 2642
```

</details>

**Comment by rarkins** [comment]
`--autodiscover` is going to autodiscover. Better you don't configure contradicting configuration options like autodiscover+repositories.

There also exists `--dry-run`. Is that what you mean by `--what-if`?

It's not really easy to answer you when you jam so many comments, questions and complaints into a wall of text, so I'll stop here.

**Reply by jessehouwing** [reply]
Sorry, I don't mean to overflow y'all with complaints. But I am trying to make things work and to explain my process based on the docs I could find. And it's quite a bit harder than the docs make it appear initially. Ive had to dig through issues, accidentally triggered massive jobs and used contradicting options without renovate telling me I'm stupid :D.

I'm far from a renovate expert, but I do understand the underlying domain. In other issues and in the docs things like "just run `npx renovate ...`" sent me down a wild goosechase.

`--autodiscover` I'm no longer using. ✅
`--dry-run` does what I was looking for. ✅

In what format would suggestions to improve docs or change defaults be most useful to you?

**Reply by rarkins** [reply]
Pull Requests would be best - all the docs are in this repo. Even if we don't accept your suggestion verbatim, you will be highlighting areas of confusion and we can propose a different wording to clarify.
